src/Rules/Test.hs

```diff
@@ -28,45 +28,116 @@ testRules = do
+            , ifMinGhcVer "711" "-fshow-warning-groups"
+            , ifMinGhcVer "801" "-fdiagnostics-color=never"
+            , ifMinGhcVer "801" "-fno-diagnostics-show-caret"
+            , pure "-dno-debug-output"
             ]
 
-            -- , "--config", "hp2ps="    ++ quote ("hp2ps")
-            -- , "--config", "hpc="      ++ quote ("hpc")
-            -- , "--config", "gs=$(call quote_path,$(GS))"
-            -- , "--config", "timeout_prog=$(call quote_path,$(TIMEOUT_PROGRAM))"
+        -- See: https://github.com/ghc/ghc/blob/master/testsuite/mk/test.mk#L291
+        let timeout_prog = "testsuite/timeout/install-inplace/bin/timeout"
+
+        quietly . cmd "python3" $
+            [ "testsuite/driver/runtests.py" ] ++
+            ["--rootdir=" ++ ("testsuite" -/- "tests")] ++
+            map ("--rootdir=" ++) tests ++
+            extraArgs ++
+            [ "-e", "windows=" ++ show windows
+            , "-e", "darwin=" ++ show darwin
+            , "-e", "config.speed=2"                        -- Use default value in GHC's test.mk
+            , "-e", "config.local=True"
+            , "-e", "config.cleanup=False"                  -- Don't clean up.
+            , "-e", "config.compiler_debugged=" ++ quote (yesNo debugged)
+            , "-e", "ghc_debugged=" ++ quote (yesNo debugged)
+            , "-e", "ghc_with_native_codegen=" ++ zeroOne withNativeCodeGen
+
+            , "-e", "config.have_interp=" ++ show withInterpreter
+            , "-e", "config.unregisterised=" ++ show unregisterised
+
+            , "-e", "ghc_compiler_always_flags=" ++ quote (unwords ghcFlags)
+            , "-e", "ghc_with_vanilla=1"                    -- TODO: do we always build vanilla?
+            , "-e", "ghc_with_dynamic=0"                    -- TODO: support dynamic
+            , "-e", "ghc_with_profiling=0"                  -- TODO: support profiling
+
+            , "-e", "config.have_vanilla=1"                 -- TODO: support other build context
+            , "-e", "config.have_dynamic=0"                 -- TODO: support dynamic
+            , "-e", "config.have_profiling=0"               -- TODO: support profiling
+            , "-e", "ghc_with_smp=" ++ zeroOne withSMP
+            , "-e", "ghc_with_llvm=0"                       -- TODO: support LLVM
+
+            , "-e", "ghc_with_threaded_rts=0"               -- TODO: support threaded
+            , "-e", "ghc_with_dynamic_rts=0"                -- TODO: support dynamic
+            , "-e", "config.ghc_dynamic_by_default=False"   -- TODO: support dynamic
+            , "-e", "config.ghc_dynamic=False"              -- TODO: support dynamic
+
+            , "-e", "config.in_tree_compiler=True"          -- Use default value, see https://github.com/ghc/ghc/blob/master/testsuite/mk/boilerplate.mk
+
+            , "--config-file=testsuite/config/ghc"
+            , "--config", "compiler="     ++ show (top -/- compiler)
```

Comment:
These two lines have been a problem for me, as the config file seems to take priority and it also defines a `config.compiler` field. @sighingnow Can you confirm this is not the case for you and the correct GHC is picked?

Reply:
I have confirm that this line let me execute the right `ghc` (inplace/bin/ghc-stage2) to compile the test.

Without this config, the `ghc` in `$PATH` will be called.

Reply:
OK, somehow the `--config` options were not working for me, glad you got that to work!